pkg/mic/mic.go

```diff
@@ -223,9 +235,20 @@ func (c *Client) Sync(exit <-chan struct{}) {
 		if deleteList != nil && len(*deleteList) > 0 {
 			beginDeletion := time.Now()
 			workDone = true
+
+			vmssNodes, err := c.getVMSSNodes(nodeRefs)
+			if err != nil {
+				glog.Error(err)
+				continue
+			}
+
 			for _, delID := range *deleteList {
 				glog.V(5).Infof("Deletion of id: %s", delID.Name)
-				inUse := c.checkIfInUse(delID, newAssignedIDs)
+				inUse, err := c.checkIfInUse(delID, newAssignedIDs, vmssNodes)
+				if err != nil {
+					glog.Error(err)
```

Comment:
We should delete this specific id from the del list, otherwise we will delete identities which are still in use, because we failed a cloud call.

Reply:
Will update.

Reply:
Wait, this is where we are doing the deletion, there shouldn't be any accidental deletions.